pkg/mattermost/mattermost_v1beta.go

```diff
@@ -178,8 +178,20 @@ func GenerateDeploymentV1Beta(mattermost *mmv1beta.Mattermost, db DatabaseConfig
 	envVarDB := db.EnvVars(mattermost)
 	initContainers := db.InitContainers(mattermost)
 
+	// Base volumes
+	volumes := mattermost.Spec.Volumes
+	volumeMounts := mattermost.Spec.VolumeMounts
+
 	// File Store
-	envVarFileStore := fileStoreEnvVars(fileStore)
+	envVarFileStore := []corev1.EnvVar{}
+	if mattermost.Spec.FileStore.Local != nil && mattermost.Spec.FileStore.Local.Enabled {
+		envVarFileStore = localFileEnvVars(mmv1beta.DefaultLocalFilePath)
+		vMount, volume := mattermostLocalFileStoreConfig(mattermost.Name)
+		volumeMounts = append(volumeMounts, vMount)
+		volumes = append(volumes, volume)
+	} else {
+		envVarFileStore = s3EnvVars(fileStore)
+	}
```

Comment:
I feel like this part becomes a bit complicated, as well as passing empty `LocalFileStore` to the `FileStoreInfo`.

Since things are quite a bit different now for different file stores, I wonder if we should not hide it behind interface to not leak that logic in multiple places, similarly as we do for `DatabaseConfig` (https://github.com/mjnagel/mattermost-operator/blob/729370d6270fac866cd69fea844f01714d98659d/pkg/mattermost/mattermost_v1beta.go#L25-L28).

We would just have methods that return configs specific for different file stores and we avoid additional `if else` logic here.

I am thinking of something like (in pseudo code):
```
type FileStoreConfig interface {
    InitContainers(mm *v1beta1.Mattermost) []InitContainer
    Volumes(mm *v1beta1.Mattermost) ([]Volumes, []VolumeMount
    EnvVars(mm *v1beta1.Mattermost) []EnvVar
}
```
So that would mean separate `struct` for each implementation. We can perhaps use `FileStoreInfo` as a field for MinIO and S3 implementations to reuse bunch of this logic, but the Local one would have a different representation.

Let me know what do you think about this approach, and feel free to ask any follow up questions if I phrased something not clear.

Reply:
It did seem weird to pass the empty `LocalFileStore`, but that seemed to be the best way to fit the current structure. More than happy to take a stab at restructuring a bit based on your comments, that makes a lot of sense. I'll reach out if I get lost or confused on anything.

Reply:
Awesome, thanks a lot!

Reply:
Sorry for the delay on this, been caught up with some other things and haven't gotten back to these changes. I started the restructure but got a little lost - will try and get back to working on this and reach out with questions.

Reply:
@Szymongib updated! It was a lot more changes than I initially thought but I think follows the spirit of what you were asking for? Let me know if any other changes are needed.